actionpack/lib/action_controller/api.rb

```diff
@@ -0,0 +1,157 @@
+  #     def index
+  #       @posts = Post.all
+  #       render json: @posts
+  #     end
+  #   end
+  #
+  # Request, response and parameters objects all work the exact same way as
+  # <tt>ActionController::Base</tt>.
+  #
+  # == Renders
+  #
+  # The default API Controller stack includes all renderers, which means you
+  # can use <tt>render :json</tt> and brothers freely in your controllers. Keep
+  # in mind that templates are not going to be rendered, so you need to ensure
+  # your controller is calling either <tt>render</tt> or <tt>redirect</tt> in
+  # all actions.
+  #
+  #   def show
+  #     @post = Post.find(params[:id])
+  #     render json: @post
+  #   end
+  #
+  # == Redirects
+  #
+  # Redirects are used to move from one action to another. You can use the
+  # <tt>redirect</tt> method in your controllers in the same way as
+  # <tt>ActionController::Base</tt>. For example:
+  #
+  #   def create
+  #     redirect_to root_url and return if not_authorized?
+  #     # do stuff here
+  #   end
+  #
+  # == Adding new behavior
+  #
+  # In some scenarios you may want to add back some functionality provided by
+  # <tt>ActionController::Base</tt> that is not present by default in
+  # <tt>ActionController::API</tt>, for instance <tt>MimeResponds</tt>. This
+  # module gives you the <tt>respond_to</tt> and <tt>respond_with</tt> methods.
+  # Adding it is quite simple, you just need to include the module in a specific
+  # controller or in <tt>ApplicationController</tt> in case you want it
+  # available to your entire app:
+  #
+  #   class ApplicationController < ActionController::API
+  #     include ActionController::MimeResponds
+  #   end
+  #
+  #   class PostsController < ApplicationController
+  #     respond_to :json, :xml
+  #
+  #     def index
+  #       @posts = Post.all
+  #       respond_with @posts
+  #     end
+  #   end
+  #
+  # Quite straightforward. Make sure to check <tt>ActionController::Base</tt>
+  # available modules if you want to include any other functionality that is
+  # not provided by <tt>ActionController::API</tt> out of the box.
+  class API < Metal
+    abstract!
+
+    module Compatibility
+      def cache_store; end
+      def cache_store=(*); end
+      def assets_dir=(*); end
+      def javascripts_dir=(*); end
+      def stylesheets_dir=(*); end
+      def page_cache_directory=(*); end
+      def asset_path=(*); end
+      def asset_host=(*); end
+      def relative_url_root=(*); end
+      def perform_caching=(*); end
+      def helpers_path=(*); end
+      def allow_forgery_protection=(*); end
+      def helper_method(*); end
+      def helper(*); end
+    end
+
+    extend Compatibility
+
+    # Shortcut helper that returns all the ActionController::API modules except the ones passed in the argument:
+    #
+    #   class MetalController
+    #     ActionController::API.without_modules(:Redirecting, :DataStreaming).each do |left|
+    #       include left
```

Comment:
Why leaving the include for the users since it is only what they can do? I think we should implement it for ourself and just require users to do:

```ruby
ActionController::API.without_modules(:Redirecting, :DataStreaming)
```